### PR TITLE
Update to React 18

### PR DESCRIPTION
**Pull request by kanekotic**
Updating to be used in react 18. Following #174 .

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,8 +55,8 @@
         "@mui/material": "^5.4.1",
         "@mui/system": "^5.4.1",
         "framer-motion": "^4.1.17",
-        "react": "^17.0.2",
-        "react-dom": "^17.0.2"
+        "react": "^18.0.0",
```
Comment (awreese) [inline package.json:58]: Retain React 17 compatibility.
```suggestion
        "react": "^17.0.2 || ^18.0.0",
```

File: package.json [hunk 2]
```diff
@@ -55,8 +55,8 @@
...
+        "react-dom": "^18.0.0"
```
Comment (awreese) [inline package.json:59]:
```suggestion
        "react-dom": "^17.0.2 || ^18.0.0"
```

File: package.json [hunk 3]
```diff
@@ -73,7 +73,7 @@
         "@mui/icons-material": "^5.0.0",
         "@mui/material": "^5.0.0",
         "@mui/system": "^5.0.0",
-        "react": "^17.0.1",
-        "react-dom": "^17.0.1"
+        "react": "^18.0.0",
```
Comment (awreese) [inline package.json:76]:
```suggestion
        "react": "^17.0.1 || ^18.0.0",
```

File: package.json [hunk 4]
```diff
@@ -73,7 +73,7 @@
...
+        "react-dom": "^18.0.0"
```
Comment (awreese) [inline package.json:77]:
```suggestion
        "react-dom": "^17.0.1 || ^18.0.0"
```

**Review by awreese — COMMENTED** [review]
Try to retain React 17 compatibility instead of forcing consumers to update React.

**Comment by ximex** [comment]
created PR for v17 and v18: https://github.com/Learus/react-material-ui-carousel/pull/177

**Pull request merged**